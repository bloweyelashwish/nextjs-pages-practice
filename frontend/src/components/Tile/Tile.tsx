import { FC, PropsWithChildren, ReactNode } from 'react';
import styled from '@emotion/styled';

import { boxShadow, borderRadius } from '~/styles/common';

export interface TileProps extends PropsWithChildren {
  header: ReactNode;
}

const Section = styled.section`
  ${({ theme }) =>
    boxShadow(theme.components.shadow1, theme.components.shadow2)};
  ${borderRadius()};

  color: ${({ theme }) => theme.font.regular};
  background-color: ${({ theme }) => theme.background};
  padding: 1vmin 4vmin 4vmin;
`;

export const Tile: FC<TileProps> = ({ header, children, ...rest }) => (
  <Section {...rest}>
    <h2>{header}</h2>
    {children}
  </Section>
);
