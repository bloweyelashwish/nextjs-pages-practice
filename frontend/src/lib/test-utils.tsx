import React from 'react';
import type { FC, PropsWithChildren, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';

import { Themes } from '~/styles/themes';

const ThemeWrapper: FC<PropsWithChildren> = ({ children }) => (
  <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
);

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: ThemeWrapper, ...options });

export * from '@testing-library/react';
export { customRender as render };
