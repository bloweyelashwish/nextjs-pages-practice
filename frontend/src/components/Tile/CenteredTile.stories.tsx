import { Meta, StoryObj } from '@storybook/react';
import { expect } from '@storybook/jest';
import { screen } from '@storybook/testing-library';

import { CenteredTile } from './CenteredTile';

const meta: Meta<typeof CenteredTile> = {
  tags: ['autodocs'],
  title: 'Content/Tile',
  component: CenteredTile,
};

export default meta;

type Story = StoryObj<typeof CenteredTile>;

export const BasicCenteredTile: Story = {
  play: () => {
    expect(screen.getByRole('heading')).toBeInTheDocument();
  },
  args: {
    header: 'Basic Centered Tile',
    children:
      'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quis quasi impedit corrupti totam quas tenetur voluptas, unde similique ipsam ad?',
  },
};
