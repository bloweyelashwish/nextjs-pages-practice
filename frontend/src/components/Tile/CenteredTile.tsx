import { FC } from 'react';
import styled from '@emotion/styled';
import { css } from '@emotion/react';

import { Tile, TileProps } from './Tile';

const generalStyles = css`
  display: flex;
  justify-content: center;
  align-items: center;
`;

const TileWrapper = styled.div`
  ${generalStyles};
  flex-flow: column;
`;

const StyledTile = styled(Tile)`
  ${generalStyles};
  flex-flow: column;
`;

export const CenteredTile: FC<TileProps> = ({ children, header, ...rest }) => {
  return (
    <TileWrapper {...rest}>
      <StyledTile header={header}>{children}</StyledTile>
    </TileWrapper>
  );
};
