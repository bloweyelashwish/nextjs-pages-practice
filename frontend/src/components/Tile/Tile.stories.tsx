import { Meta, StoryObj } from '@storybook/react';
import { expect } from '@storybook/jest';
import { screen } from '@storybook/testing-library';

import { Tile } from './Tile';

const meta: Meta<typeof Tile> = {
  tags: ['autodocs'],
  title: 'Content/Tile',
  component: Tile,
};

export default meta;

type Story = StoryObj<typeof Tile>;

export const BasicTile: Story = {
  play: () => {
    expect(screen.getByRole('heading')).toBeInTheDocument();
  },
  args: {
    header: 'Basic Tile',
    children:
      'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quis quasi impedit corrupti totam quas tenetur voluptas, unde similique ipsam ad?',
  },
};
