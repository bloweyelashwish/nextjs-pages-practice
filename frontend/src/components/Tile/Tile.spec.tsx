import { render } from '~/lib/test-utils';

import { Tile } from './Tile';

describe('Tile test cases', () => {
  it('Tile renders on the page', () => {
    const { asFragment } = render(
      <Tile header="Tile header">
        Lorem ipsum dolor sit amet consectetur, adipisicing elit. Doloremque
        cumque ad corrupti? Maxime vero dolor quibusdam eius corporis.
        Asperiores, nemo!
      </Tile>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
