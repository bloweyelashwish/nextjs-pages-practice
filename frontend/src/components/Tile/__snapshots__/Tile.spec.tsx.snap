// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile test cases Tile renders on the page 1`] = `
<DocumentFragment>
  <section
    class="css-glfd8h"
  >
    <h2>
      Tile header
    </h2>
    Lorem ipsum dolor sit amet consectetur, adipisicing elit. Doloremque cumque ad corrupti? Maxime vero dolor quibusdam eius corporis. Asperiores, nemo!
  </section>
</DocumentFragment>
`;
